// @netbasal/ngx-content-loader/esm5/lib/content-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * @return {?}
 */
function uid() {
    return Math.random()
        .toString(36)
        .substring(2);
}
var ContentLoaderComponent = /** @class */ (function () {
    function ContentLoaderComponent() {
        this.animate = true;
        this.baseUrl = '';
        this.width = 400;
        this.height = 130;
        this.speed = 2;
        this.preserveAspectRatio = 'xMidYMid meet';
        this.primaryColor = '#f9f9f9';
        this.secondaryColor = '#ecebeb';
        this.primaryOpacity = 1;
        this.secondaryOpacity = 1;
        this.idClip = uid();
        this.idGradient = uid();
        this.defautlAnimation = ['-3; 1', '-2; 2', '-1; 3'];
        this.rtlAnimation = ['1; -3', '2; -2', '3; -1'];
    }
    /**
     * @return {?}
     */
    ContentLoaderComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.animationValues = this.rtl ? this.rtlAnimation : this.defautlAnimation;
    };
    Object.defineProperty(ContentLoaderComponent.prototype, "fillStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return {
                fill: "url(" + this.baseUrl + "#" + this.idGradient + ")"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ContentLoaderComponent.prototype, "clipStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return "url(" + this.baseUrl + "#" + this.idClip + ")";
        },
        enumerable: true,
        configurable: true
    });
    ContentLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'content-loader',
                    template: "<svg attr.viewBox=\"0 0 {{width}} {{height}}\" [attr.preserveAspectRatio]=\"preserveAspectRatio\" [ngStyle]=\"style\">\n\n  <rect [attr.clip-path]=\"clipStyle\" [ngStyle]=\"fillStyle\" x=\"0\" y=\"0\" [attr.width]=\"width\" [attr.height]=\"height\" />\n\n  <defs>\n\n    <clipPath [attr.id]=\"idClip\">\n      <ng-content></ng-content>\n    </clipPath>\n\n    <linearGradient [attr.id]=\"idGradient\">\n\n      <stop offset=\"0%\" [attr.stop-color]=\"primaryColor\" [attr.stop-opacity]=\"primaryOpacity\">\n        <ng-container *ngIf=\"animate\">\n          <animate attributeName=\"offset\" [attr.values]=\"animationValues[0]\" attr.dur={{speed}} s repeatCount=\"indefinite\" />\n        </ng-container>\n      </stop>\n\n      <stop offset=\"50%\" [attr.stop-color]=\"secondaryColor\" [attr.stop-opacity]=\"secondaryOpacity\">\n        <ng-container *ngIf=\"animate\">\n          <animate attributeName=\"offset\" [attr.values]=\"animationValues[1]\" attr.dur={{speed}} s repeatCount=\"indefinite\" />\n        </ng-container>\n      </stop>\n\n      <stop offset=\"100%\" [attr.stop-color]=\"primaryColor\" [attr.stop-opacity]=\"primaryOpacity\">\n        <ng-container *ngIf=\"animate\">\n          <animate attributeName=\"offset\" [attr.values]=\"animationValues[2]\" attr.dur={{speed}} s repeatCount=\"indefinite\" />\n        </ng-container>\n      </stop>\n\n    </linearGradient>\n\n  </defs>\n\n</svg>\n"
                }] }
    ];
    ContentLoaderComponent.propDecorators = {
        animate: [{ type: Input }],
        baseUrl: [{ type: Input }],
        width: [{ type: Input }],
        height: [{ type: Input }],
        speed: [{ type: Input }],
        preserveAspectRatio: [{ type: Input }],
        primaryColor: [{ type: Input }],
        secondaryColor: [{ type: Input }],
        primaryOpacity: [{ type: Input }],
        secondaryOpacity: [{ type: Input }],
        uniqueKey: [{ type: Input }],
        rtl: [{ type: Input }],
        style: [{ type: Input }]
    };
    return ContentLoaderComponent;
}());
export { ContentLoaderComponent };
if (false) {
    /** @type {?} */
    ContentLoaderComponent.prototype.animate;
    /** @type {?} */
    ContentLoaderComponent.prototype.baseUrl;
    /** @type {?} */
    ContentLoaderComponent.prototype.width;
    /** @type {?} */
    ContentLoaderComponent.prototype.height;
    /** @type {?} */
    ContentLoaderComponent.prototype.speed;
    /** @type {?} */
    ContentLoaderComponent.prototype.preserveAspectRatio;
    /** @type {?} */
    ContentLoaderComponent.prototype.primaryColor;
    /** @type {?} */
    ContentLoaderComponent.prototype.secondaryColor;
    /** @type {?} */
    ContentLoaderComponent.prototype.primaryOpacity;
    /** @type {?} */
    ContentLoaderComponent.prototype.secondaryOpacity;
    /** @type {?} */
    ContentLoaderComponent.prototype.uniqueKey;
    /** @type {?} */
    ContentLoaderComponent.prototype.rtl;
    /** @type {?} */
    ContentLoaderComponent.prototype.style;
    /** @type {?} */
    ContentLoaderComponent.prototype.idClip;
    /** @type {?} */
    ContentLoaderComponent.prototype.idGradient;
    /** @type {?} */
    ContentLoaderComponent.prototype.defautlAnimation;
    /** @type {?} */
    ContentLoaderComponent.prototype.rtlAnimation;
    /** @type {?} */
    ContentLoaderComponent.prototype.animationValues;
}
